Packs/Campaign/Scripts/PerformActionOnCampaignIncidents/PerformActionOnCampaignIncidents.py

```diff
@@ -2,201 +2,519 @@
+    ids_to_add_to_campaign = list(set(ids) - set(campaign_incidents_ids))
+    if not ids_to_add_to_campaign:
+        return "No new incidents to add to campaign."
+    involved_incidents_count = int(demisto.dt(campaign_context, "Contents.context.EmailCampaign.involvedIncidentsCount")[0])
+    involved_incidents_count += len(ids_to_add_to_campaign)
+
+    campaign_incidents_context = [_parse_incident_context_to_valid_incident_campaign_context(
+        id, fields_to_display) for id in ids_to_add_to_campaign]
+    for id in ids_to_add_to_campaign:
+        _set_part_of_campaign_filed(id, campaign_id)
+    _set_incidents_to_campaign(campaign_id, campaign_incidents_context, True)
+    _link_or_unlink_between_incidents(campaign_id, ids_to_add_to_campaign, "link")
+    _set_involved_incidents_count(campaign_id, involved_incidents_count)
+
+    return COMMAND_SUCCESS.format(action="added to campaign", ids=ids_to_add_to_campaign)
+
+
+def perform_remove_from_campaign(ids: list[str], action: str) -> str:
+    """Removes incidents from an email campaign.
+
+    Args:
+        ids (list[str]): The IDs of the incidents to remove. Required.
+        action (str): The action that was performed to trigger this, e.g. "unlink". 
+            Required.
+
+    Returns:
+        str: A message indicating success.
+
+    Steps:
+        1. Get the current campaign context
+        2. Get the existing campaign incident IDs
+        3. Calculate new count of involved incidents
+        4. Filter incidents to remove
+        5. Update campaign metadata
+        6. Update links between incidents
+        7. Return success message
+    """
+    campaign_id = demisto.incident()["id"]
+    campaign_context = _get_context(campaign_id)
+    campaign_incidents_context = demisto.dt(campaign_context, "Contents.context.EmailCampaign.incidents") or []
+
+    campaign_incidents_ids = [incident["id"] for incident in campaign_incidents_context]
+    ids_to_remove_from_campaign = list(set(ids) & set(campaign_incidents_ids))
+    if not ids_to_remove_from_campaign:
+        return "No incidents to remove from the campaign."
+    involved_incidents_count = int(demisto.dt(campaign_context, "Contents.context.EmailCampaign.involvedIncidentsCount")[0])
+    involved_incidents_count -= len(ids_to_remove_from_campaign)
+    campaign_incidents_context = [
+        incident for incident in campaign_incidents_context if incident['id'] not in ids_to_remove_from_campaign]
+
+    for id_ in ids_to_remove_from_campaign:
+        _set_part_of_campaign_filed(id_, "None")
+    _set_incidents_to_campaign(campaign_id, campaign_incidents_context)
+    _link_or_unlink_between_incidents(campaign_id, ids_to_remove_from_campaign, "unlink")
+    _set_involved_incidents_count(campaign_id, involved_incidents_count)
+
+    return COMMAND_SUCCESS.format(action="removed from campaign", ids=ids_to_remove_from_campaign)
+
+
+def perform_reopen(ids: list[str], action: str) -> str:
```

Comment:
is there any possible `action` value other than `"reopen"`? if not, let's omit this argument